src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -431,30 +433,64 @@ private RestAzureNS.AzureOperationResponse<ProtectionPolicyResource> CreateorMod
+                {
+                    // copy the backupSchedule time to retentionPolicy after converting to UTC
+                    AzureWorkloadProviderHelper.CopyScheduleTimeToRetentionTimes(
+                        (CmdletModel.VaultRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy,
+                        (CmdletModel.SimpleSchedulePolicy)((AzureFileSharePolicy)policy).SchedulePolicy);
+                    Logger.Instance.WriteDebug("Copy of RetentionTime from with SchedulePolicy to RetentionPolicy is successful");
+
+                    // Now validate both RetentionPolicy and SchedulePolicy matches or not
+                    PolicyHelpers.ValidateLongTermRetentionPolicyWithSimpleSchedulePolicy(
+                        (CmdletModel.VaultRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy,
+                        (CmdletModel.SimpleSchedulePolicy)((AzureFileSharePolicy)policy).SchedulePolicy);
+                    Logger.Instance.WriteDebug("Validation of Retention policy with Schedule policy is successful");
+                    azureFileShareProtectionPolicy.RetentionPolicy = PolicyHelpers.GetServiceClientVaultRetentionPolicy(
+                               (CmdletModel.VaultRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy);
+                } 
+                else if (ProviderData.ContainsKey(PolicyParams.RetentionPolicy) && ProviderData[PolicyParams.RetentionPolicy] != null && ProviderData[PolicyParams.RetentionPolicy].GetType() == typeof(CmdletModel.LongTermRetentionPolicy))
+                {
+                    // copy the backupSchedule time to retentionPolicy after converting to UTC
+                    AzureWorkloadProviderHelper.CopyScheduleTimeToRetentionTimes(
+                        (CmdletModel.LongTermRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy,
+                        (CmdletModel.SimpleSchedulePolicy)((AzureFileSharePolicy)policy).SchedulePolicy);
+                    Logger.Instance.WriteDebug("Copy of RetentionTime from with SchedulePolicy to RetentionPolicy is successful");
+
+                    // Now validate both RetentionPolicy and SchedulePolicy matches or not
+                    PolicyHelpers.ValidateLongTermRetentionPolicyWithSimpleSchedulePolicy(
+                        (CmdletModel.LongTermRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy,
+                        (CmdletModel.SimpleSchedulePolicy)((AzureFileSharePolicy)policy).SchedulePolicy);
+                    Logger.Instance.WriteDebug("Validation of Retention policy with Schedule policy is successful");
+                    azureFileShareProtectionPolicy.RetentionPolicy = PolicyHelpers.GetServiceClientLongTermRetentionPolicy(
+                               (CmdletModel.LongTermRetentionPolicy)((AzureFileSharePolicy)policy).RetentionPolicy);
```

Comment:
this code might get reduced after we make LTR as base class for VaultRetentionPolicy

Reply:
@hiaga Refactored VaultRetentionPolicy to use LTR as base